П.П_1.1.4

Salavat1994 opened the pull request.

Salavat1994 committed:
П.П_1.1.4

pdasilem commented on .idea/.gitignore:
```diff
@@ -0,0 +1,8 @@
+# Default ignored files
```
убрать служебные файлы проекта из гита. используй гитигнор, например

Salavat1994 replied:
Выполнено

pdasilem replied:
гитигнор положи в корень проекта, удали из пр служебные файлы

pdasilem commented on src/main/java/jm/task/core/jdbc/Main.java:
```diff
@@ -1,7 +1,29 @@
 public class Main {
     public static void main(String[] args) {
         // реализуйте алгоритм здесь
+        UserServiceImpl table = new UserServiceImpl();
```
нейминг лучше всего использовать по имени класса/интерфейса. имя userService будет более наглядным.

Salavat1994 replied:
Выполнено

pdasilem replied:
я так понимаю, коммит еще не сделан?

pdasilem commented on src/main/java/jm/task/core/jdbc/Main.java:
```diff
@@ -1,7 +1,29 @@
+
+import java.util.List;
+
+import static com.sun.xml.fastinfoset.alphabet.BuiltInRestrictedAlphabets.table;
```
это для чего?

Salavat1994 replied:
Удалено

pdasilem commented on src/main/java/jm/task/core/jdbc/dao/UserDaoHibernateImpl.java:
```diff
@@ -1,42 +1,93 @@
+            session.beginTransaction();
+            session.createSQLQuery(query).executeUpdate();
+            session.getTransaction().commit();
+        } catch (SQLException e) {
```
где роллбэк транзакции при ошибке? и так во всех методах

Salavat1994 replied:
Реализовано

pdasilem commented on src/main/java/jm/task/core/jdbc/dao/UserDaoHibernateImpl.java:
```diff
@@ -1,42 +1,93 @@
+            session.getTransaction().commit();
+        } catch (SQLException e) {
+            e.printStackTrace();
+            return null;
```
зачем?

pdasilem commented on src/main/java/jm/task/core/jdbc/model/User.java:
```diff
@@ -1,12 +1,12 @@
+import javax.persistence.*;
 
+@Entity(name = "Users")
 @Table
```
обязательна ли эта аннотация? Если ты ее поставил - для чего?

pdasilem commented on src/main/java/jm/task/core/jdbc/service/UserServiceImpl.java:
```diff
@@ -1,31 +1,36 @@
 public class UserServiceImpl implements UserService {
-    public void createUsersTable() {
 
+    UserDao userDao = new UserDaoHibernateImpl();
```
все поля классов приватные финальные

Salavat1994 replied:
Выполнено

Salavat1994 committed:
Add git ignore

Salavat1994 committed:
Поменял имя переменной, добавил таргет в гитИгнор

Salavat1994 committed:
Поменял имя переменной, добавил таргет в гитИгнор
Добавил к @Table имя
Проинициализировал List
Сделал userDao приватным и финальным

Salavat1994 committed:
Поменял имя переменной, добавил таргет в гитИгнор
Добавил к @Table имя
Проинициализировал List
Сделал userDao приватным и финальным
Создал транзакцию

Salavat1994 committed:
Поменял имя переменной, добавил таргет в гитИгнор
Добавил к @Table имя
Проинициализировал List
Сделал userDao приватным и финальным
Создал транзакцию

Salavat1994 committed:
КОММИТ 1

Salavat1994 committed:
Delete hs_err_pid13320.log

Salavat1994 committed:
Delete replay_pid13320.log

Salavat1994 committed:
Delete hs_err_pid14724.log

Salavat1994 committed:
Delete replay_pid14724.log

Salavat1994 committed:
Commit_1

Salavat1994 committed:
Merge remote-tracking branch 'origin/P.P_1.1.4' into P.P_1.1.4

pdasilem commented on src/main/java/jm/task/core/jdbc/Main.java:
```diff
@@ -1,7 +1,29 @@
 public class Main {
     public static void main(String[] args) {
         // реализуйте алгоритм здесь
+        UserServiceImpl userService = new UserServiceImpl();
```
внедряй через интерфейсы

pdasilem commented on pom.xml:
```diff
@@ -31,9 +31,9 @@
-            <version>8.0.25</version>
+            <groupId>org.postgresql</groupId>
+            <artifactId>postgresql</artifactId>
+            <version>42.6.0</version>
```
версия?

Salavat1994 committed:
Delete TaskJDBC.iml

Salavat1994 committed:
Commit_actual

Salavat1994 committed:
Merge remote-tracking branch 'origin/P.P_1.1.4' into P.P_1.1.4
